### DISCUSSION TITLE
HM-ES-TX-WM mit einem verschlüsselten Smart Meter

### DISCUSSION THREAD

**Discussion by friedpa** [OpenCCU/OpenCCU#2077] [🇩🇪 Fragen&Antworten] [answered]
Hallo,
ich würde gerne meinen jetzt installierten Smart Meter mit meinem HM-ES-TX-WM auslesen. Ich habe mir bereits den IEC Auslesekopf besorgt und leider zu spät mitbekommen, dass der Smart Meter die Daten nur verschlüsselt ausgibt. Gibt es eine Möglichkeit die Daten in einem Script zu dechiffrieren? Den AES Key habe ich bekommen.
Viele Dank im Voraus für jeden Tipp.

**Comment by jens-maus** [comment, marked as answer]
Meines Wissen nicht. der HM-ES-TX-WM arbeitet komplett transparent. Das Protokoll das über den IEC Auslesekopf abgegriffen wird, wird komplett in der Firmware des HM-ES-TX-WM verarbeitet und dann lediglich nach außen (d.h. für die CCU) via Kanäle, etc. zur Verfügung gestellt. D.h. im Grunde müsste man Zugriff auf die Firmware des HM-ES-TX-WM haben. Diese ist aber komplett closed-source und nur für eQ3/ELV zu ändern bzw. einzusehen.

Um deinen Smart Meter also mit dem HM-ES-TX-WM auslesen zu können müsstest du Kontakt mit eQ3/ELV aufnehmen und diese davon überzeugen die entsprechenden Entschlüsselungsroutinen, etc. dort zu integrieren. Sicherlich kein unmögliches Unterfangen, aber auch recht unwahrscheinlich das eQ3/ELV hier für einen einzelnen Nutzer solche Anpassungen vornimmt.

**Comment by friedpa** [comment]
Hallo Jens,
danke für die schnelle Antwort.
Vielleicht finden sich ja noch Mitstreiter, da ALLE Smart Meter in Österreich verschlüsselt sind.
LG Paul

**Reply by jens-maus** [reply]
> Vielleicht finden sich ja noch Mitstreiter, da ALLE Smart Meter in Österreich verschlüsselt sind.

Interessant. Wusste gar nicht das die österreichischen Datenschützer **noch** paranoider sind als die deutschen Datenschützer bzw. eine noch größere Lobby haben in Ihrem Land :)

**Comment by friedpa** [comment]
Ich habe eine Anfrage an eQ3 geschickt und werde berichten.

**Comment by friedpa** [comment]
Hallo Gents,
ich habe mehrmals an den Support geschrieben und bis heute keine Rückmeldung bekommen.
Vielleicht kann ja mal einer von Euch mit mehr "Gewicht" bei eQ3/ELV nachfragen, es betrifft ja den gesamten österreichischen Markt, ob man nun den HM-ES-TX-WM kübeln muss, oder doch vielleicht das Thema noch aufgegriffen wird.
Vielen Dank!

**Reply by jens-maus** [reply]
Ich denke – aus meiner Erfahrung heraus – wirst du hier vmtl. nicht weiterkommen. Ich halte es für höchst unwahrscheinlich das ELV hier die notwendigen Entschlüsslungsroutinen in die Firmware des HM-ES-TX-WM noch packen wird. Abgesehen davon das das ziemlicher Aufwand wäre und vmtl. in der Firmware vllt. sogar gar nicht mehr Platz für solche Modifikationen ist, ist die BidCos-RF Welt von eQ3/ELV ohnehin schon mehr oder weniger abgekündigt und wenn dann wird es vielleicht irgendwann mal eine homematicIP Variante des Selbigen geben.

Ich würde an deiner Stelle also nicht davon ausgehen das hier irgendetwas von eQ3/ELV zu diesem speziellen Thema bzw. zu diesem alten BidCos-RF Gerät kommen wird. Viel mehr Chancen solltest du damit haben etwaige Homebrewgeräte einzusetzen (wie z.B. https://homematic-forum.de/forum/viewtopic.php?f=76&t=77690).
